frontend/src/components/upload/FileList.tsx

```diff
@@ -1,41 +1,99 @@
 import { ActionIcon, Table } from "@mantine/core";
-import { Dispatch, SetStateAction } from "react";
 import { TbTrash } from "react-icons/tb";
-import { FileUpload } from "../../types/File.type";
+import { GrUndo } from "react-icons/gr";
+import { FileListItem } from "../../types/File.type";
 import { byteToHumanSizeString } from "../../utils/fileSize.util";
 import UploadProgressIndicator from "./UploadProgressIndicator";
 import { FormattedMessage } from "react-intl";
 
-const FileList = ({
+const FileListRow = ({
+  file,
+  onRemove,
+  onRestore,
+}: {
+  file: FileListItem,
+  onRemove?: () => void,
+  onRestore?: () => void,
+}) => {
+  {
+    const uploadable = "uploadingProgress" in file
+    const uploading = uploadable && file.uploadingProgress !== 0
+    const removable = uploadable ? file.uploadingProgress === 0 : onRemove && !file.deleted;
+    const restorable = onRestore && !uploadable && !!file.deleted; // maybe undefined, force boolean
+    const deleted = !uploadable && !!file.deleted;
+
+    return  <tr style={{
+      color: deleted ? "rgba(120, 120, 120, 0.5)" : "inherit",
+      textDecoration: deleted ? "line-through" : "none",
+    }}>
+      <td>{file.name}</td>
+      <td>{byteToHumanSizeString(+file.size)}</td>
+      <td>
+        {
+          removable && (
+            <ActionIcon
+              color="red"
+              variant="light"
+              size={25}
+              onClick={onRemove}
+            >
+              <TbTrash />
+            </ActionIcon>
+          )
+        }
+        {
+          uploading && <UploadProgressIndicator progress={file.uploadingProgress} />
+        }
+        {
+          restorable && (
+            <ActionIcon
+              color="blue"
```

Comment:
```suggestion
              color="primary"
```
I prefer the primary Pingvin Share color but let me know if you don't agree.